framework/db/Query.php

```diff
@@ -387,7 +387,19 @@ public function min($q, $db = null)
      */
     public function max($q, $db = null)
     {
-        return $this->queryScalar("MAX($q)", $db);
+        return $this->toInt($this->queryScalar("MAX($q)", $db));
+    }
+
+    /**
+     * Converts string value to int if possible.
+     *
+     * @param string $value
+     * @return int
+     */
+    private function toInt($value)
+    {
+        $int = (int) $value;
+        return (string) $int === $value ? $int : $value;
```

Comment:
Integer typecast may be unsafe as it may overflow PHP max int restiction.
Comparing `(string) $int === $value` is a bad idea. How it will handle values like `5E91`?